packages/api/src/controllers/playback.ts

```diff
@@ -44,11 +45,12 @@ const newPlaybackInfo = (
 const getAssetPlaybackUrl = async (
   config: Request["config"],
   ingest: string,
-  id: string
+  id: string,
+  user?: User
 ) => {
   const asset =
     (await db.asset.getByPlaybackId(id)) ??
-    (await db.asset.getByIpfsCid(id)) ??
+    (await db.asset.getByIpfsCid(id, user)) ??
     (await db.asset.getBySourceURL("ipfs://" + id)) ??
     (await db.asset.getBySourceURL("ar://" + id));
```

Comment:
There are also decentralized ID queries and also need an user-preference logic.

Reply:
You're right! added